pkg/runtime/codec.go

```diff
@@ -22,6 +22,24 @@ import (
 	"k8s.io/kubernetes/pkg/util/yaml"
 )
 
+// Encode is a convenience wrapper for encoding to a []byte from an Encoder
+// TODO: these are transitional interfaces to reduce refactor cost as Codec is altered.
+func Encode(e Encoder, obj Object) ([]byte, error) {
+	return e.Encode(obj)
+}
+
+// Decode is a convenience wrapper for decoding data into an Object.
+// TODO: these are transitional interfaces to reduce refactor cost as Codec is altered.
+func Decode(d Decoder, data []byte) (Object, error) {
+	return d.Decode(data)
+}
+
+// DecodeInto performs a Decode into the provided object.
+// TODO: these are transitional interfaces to reduce refactor cost as Codec is altered.
+func DecodeInto(d Decoder, data []byte, into Object) error {
+	return d.DecodeInto(data, into)
```

Comment:
All of the decode variants have been removed and replaced (in my pr) with
something slightly more flexible than DecodeIntoWithSpecificVersionKind.
There will be one decode and one encode method, taking options on decoding.

On Dec 11, 2015, at 7:34 PM, Chao Xu <notifications@github.com> wrote:

In pkg/runtime/codec.go
<https://github.com/kubernetes/kubernetes/pull/18487#discussion_r47421872>:

> +// Encode is a convenience wrapper for encoding to a []byte from an Encoder
> +// TODO: these are transitional interfaces to reduce refactor cost as Codec is altered.
> +func Encode(e Encoder, obj Object) ([]byte, error) {
> +	return e.Encode(obj)
> +}
> +
> +// Decode is a convenience wrapper for decoding data into an Object.
> +// TODO: these are transitional interfaces to reduce refactor cost as Codec is altered.
> +func Decode(d Decoder, data []byte) (Object, error) {
> +	return d.Decode(data)
> +}
> +
> +// DecodeInto performs a Decode into the provided object.
> +// TODO: these are transitional interfaces to reduce refactor cost as Codec is altered.
> +func DecodeInto(d Decoder, data []byte, into Object) error {
> +	return d.DecodeInto(data, into)

Will the interface contain DecodeInto()?
Will DecodeIntoWithSpecifiedVersionKind be implemented using d.Decode()?

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/18487/files#r47421872>.